### PR TITLE
Feature/regtest options cli wallet

### PR DESCRIPTION
**Pull request by OBorce**
- Add an option to change the regtest chain config options for the CLI wallet same as the GUI wallet and the node deamon.
- Add a call to decommission stake pool in the functional tests and check for locked balance after that.
- Refactor asserts in the wallet functional tests for better error messages if they fail.

### CODE REVIEW

File: wallet/wallet-cli-lib/tests/cli_test_framework.rs [hunk 1]
```diff
@@ -298,7 +298,22 @@ impl CliTestFramework {
         let manager_task = manager.main_in_task();
 
         let wallet_options = WalletCliArgs {
-            network: Network::Regtest,
+            network: Network::Regtest(Box::new(config::regtest_options::ChainConfigOptions {
+                chain_magic_bytes: None,
```
Comment (azarovh) [inline wallet/wallet-cli-lib/tests/cli_test_framework.rs:302]: looks like a good reason to add `ChainConfigOptions::new`

**Review by azarovh — APPROVED** [review]

File: wallet/wallet-cli-lib/tests/cli_test_framework.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply wallet/wallet-cli-lib/tests/cli_test_framework.rs:302]: I think this is fine... with this huge number of parameters, using positional arguments would be a disaster.

**Review by TheQuantumPhysicist — APPROVED** [review]

File: wallet/wallet-cli-lib/tests/cli_test_framework.rs (hunk 1 above)
Comment (azarovh) [reply wallet/wallet-cli-lib/tests/cli_test_framework.rs:302]: I meant `new()` or default

Comment (azarovh) [reply wallet/wallet-cli-lib/tests/cli_test_framework.rs:302]: can be dropped as discussed

**Pull request merged**